cit-api/pipeline/management/commands/send-user-tracking.py

```diff
@@ -0,0 +1,85 @@
+from django.core.management.base import BaseCommand
+from django.db import connection
+from datetime import date, timedelta
+import csv
+import os
+import requests
+import json
+from io import StringIO
+from requests.auth import HTTPBasicAuth
+import calendar
+import base64
+
+
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        send_tracking_email()
+
+def send_tracking_email():
+    token_request_body = {'grant_type': 'client_credentials'}
+    response = requests.post(os.environ.get("EMAIL_AUTH_HOST") + "/auth/realms/jbd6rnxw/protocol/openid-connect/token",
+                             data=token_request_body, auth=HTTPBasicAuth(os.environ.get("EMAIL_CLIENT_ID"), os.environ.get("EMAIL_CLIENT_SECRET")))
+    if response.status_code == 200:
+        access_token = response.json()["access_token"]
+        headers = {
+            "Authorization": "Bearer " + access_token,
+            "Content-Type": "application/json",
+        }
+        response = requests.get(os.environ.get("EMAIL_SERVICE_HOST") + "/api/v1/health", headers=headers)
+        # Get time span
+        today = date.today()
+        three_weeks_ago = today - timedelta(weeks=3)
+        start_end = calendar.monthrange(three_weeks_ago.year, three_weeks_ago.month)
+        from_date = f"{three_weeks_ago.year}-{three_weeks_ago.month}-1"
+        to_date = f"{three_weeks_ago.year}-{three_weeks_ago.month}-{start_end[1]}"
+        # Get user tracking records
+        csv_content = get_user_tracking(from_date, to_date)
+        encoded_csv_content = encode_csv_file(csv_content)
+        environment_level = os.environ.get("ENV_LEVEL")
+        if response.status_code == 200:
+            email_config = {
+                "bcc": [],
+                "bodyType": "html",
+                "body": construct_email_body(from_date, to_date),
+                "cc": [],
+                "delayTS": 0,
+                "encoding": "utf-8",
+                "from": os.environ.get("EMAIL_SENDING_ADDRESS"),
+                "priority": "normal",
+                "subject": construct_email_body(from_date, to_date),
+                "to": [os.environ.get("USER_TRACKING_TO_EMAIL")],
+                "tag": "CIT_Admin_Notification",
+                "attachments": [
+                    {
+                        "content": f"{encoded_csv_content}",
+                        "encoding": "base64",
+                        "filename": f"{environment_level}_cit_user_bca_access_from_{from_date}_to_{to_date}.csv"
+                    }
+                ]
+            }
+            email_config_json = json.dumps(email_config)
+            response = requests.post(os.environ.get("EMAIL_SERVICE_HOST") + "/api/v1/email", data=email_config_json, headers=headers)
+        return response
```

Comment:
We don't actually do anything with this response currently, so we should print it out  or something to make it clear that the run was a success.

Reply:
This now checked in the main function above